code/modules/mob/living/simple_animal/abnormality/aleph/white_night.dm

```diff
@@ -196,258 +91,386 @@ GLOBAL_LIST_EMPTY(apostles)
-			// The farther you are from white night - the less damage it deals
-			var/dealt_damage = max(1, holy_revival_damage - attack_range)
-			H.apply_damage(dealt_damage, PALE_DAMAGE, null, H.run_armor_check(null, PALE_DAMAGE), spread_damage = TRUE)
-			if((holy_revival_damage - attack_range) > 5)
-				H.emote("scream")
-			to_chat(H, "<span class='userdanger'>The holy light... IT BURNS!!</span>")
+/mob/living/simple_animal/hostile/abnormality/white_night/proc/revive_target(mob/living/L, attack_range = 1, faction_check = "apostle")
+	if(!(faction_check in L.faction))
+		playsound(L.loc, 'sound/machines/clockcult/ark_damage.ogg', 50 - attack_range, TRUE, -1)
+		// The farther you are from white night - the less damage it deals
+		var/dealt_damage = max(5, holy_revival_damage - attack_range)
+		L.apply_damage(dealt_damage, PALE_DAMAGE, null, L.run_armor_check(null, PALE_DAMAGE), spread_damage = TRUE)
+		if(ishuman(L) && dealt_damage > 25)
+			L.emote("scream")
+		to_chat(L, "<span class='userdanger'>The holy light... IT BURNS!!</span>")
 	else
-		if(H.stat == DEAD && H.mind && faction_check == "apostle") // No gaming
-			H.regenerate_limbs()
-			H.regenerate_organs()
-			H.dna.species.GiveSpeciesFlight(H)
-			H.revive(full_heal = TRUE, admin_revive = FALSE)
-			H.grab_ghost(force = TRUE)
-			to_chat(H, "<span class='notice'>The holy light compels you to live!</span>")
-		else
-			H.adjustStaminaLoss(-200)
-			H.adjustBruteLoss(-holy_revival_damage*5)
-			H.adjustFireLoss(-holy_revival_damage*5)
-			H.adjustSanityLoss(holy_revival_damage*5) // It actually heals, don't worry
-			H.regenerate_limbs()
-			H.regenerate_organs()
-			to_chat(H, "<span class='notice'>The holy light heals you!</span>")
-
-/mob/living/simple_animal/hostile/abnormality/white_night/proc/fire_field()
-	if(fire_field_cooldown > world.time)
-		return
-	var/turf/target_c = get_turf(src)
-	var/list/fire_zone = list()
-	fire_field_cooldown = (world.time + fire_field_cooldown_base)
-	for(var/i = 1 to field_range)
-		playsound(target_c, 'sound/machines/clockcult/stargazer_activate.ogg', 50, 1)
-		fire_zone = spiral_range_turfs(i, target_c) - spiral_range_turfs(i-1, target_c)
-		for(var/turf/open/T in fire_zone)
-			new /obj/effect/temp_visual/cult/turf/floor(T)
-		SLEEP_CHECK_DEATH(1.5)
-	SLEEP_CHECK_DEATH(3)
-	for(var/i = 1 to field_range)
-		fire_zone = spiral_range_turfs(i, target_c) - spiral_range_turfs(i-1, target_c)
-		playsound(target_c, "explosion", 25, TRUE)
-		for(var/turf/open/T in fire_zone)
-			new /obj/effect/temp_visual/fire/fast(T)
-			for(var/mob/living/L in T.contents)
-				if("apostle" in L.faction)
-					continue
-				L.apply_damage(20, PALE_DAMAGE, null, L.run_armor_check(null, PALE_DAMAGE), spread_damage = TRUE)
-				to_chat(L, "<span class='userdanger'>You're hit by [src]'s fire field!</span>")
-		SLEEP_CHECK_DEATH(1.5)
+		if(istype(L, /mob/living/simple_animal/hostile/apostle) && L.stat == DEAD)
+			L.revive(full_heal = TRUE, admin_revive = FALSE)
```

Comment:
This probably shouldn't full heal, as it kinda makes suppressing them at all entirely rewardless and just encourages quite literally putting them somewhere else. Maybe a half health revive so all the work done by agents aren't entirely instantly undone, but still enough so that they can come back from it and be a threat? Also the Apostles seem to fade on death but immediately pop back into reality when revived, can we get a downed Sprite for them?

Reply:
Something just occured to me was that you can probably gib apostles with Smile weapon. Might want a check to count apostles and spawn a new one if they get gibbed? Uncertain.

Reply:
You cannot gib apostles, at all.
And full revive will stay: You are not supposed to fight apostles, you are supposed to go for the White Night.